### PR TITLE
Makes it possible to feed xenos food.

### PR DESCRIPTION
**Pull request by lbnesquik**
## About The Pull Request
They still do not process it.

## Why It's Good For The Game
You can now roleplay having a date with a bean. That's it.

## Changelog
:cl:
qol: allowed xenos to be fed food for the sake of fluff
/:cl:

### CODE REVIEW

**Comment by ilikesaying** [comment]
they already can be fed food? i literally handfed a boiler a burger?

**Comment by lbnesquik** [comment]
> 
> 
> they already can be fed food? i literally handfed a boiler a burger?

Last game, i couldn't feed a queen drinks or borgers. Soooo, it seems like there is inconsistent code.

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Relative pathing, you tocuh it you fix it
also since when does this not work its been working

**Comment by lbnesquik** [comment]
> 
> 
> Relative pathing, you tocuh it you fix it

Reeeeeeeeeeeeee

**Comment by lbnesquik** [comment]
Alright dweebs. I implemented changes to make sure condiments work like other food sources, fixed a weird proc, some bypasses and redundant checks.

File: code/game/objects/items/reagent_containers/food/condiment.dm [hunk 1]
```diff
@@ -15,136 +15,138 @@
 	center_of_mass = list("x"=16, "y"=6)
 	volume = 50
 
-	attackby(obj/item/W as obj, mob/user as mob)
-
+/obj/item/reagent_containers/food/condiment/attackby(obj/item/W as obj, mob/user as mob)
```
Comment (SplinterGP) [inline code/game/objects/items/reagent_containers/food/condiment.dm:18]:
```suggestion
/obj/item/reagent_containers/food/condiment/attackby(obj/item/W, mob/user)
```

File: code/game/objects/items/reagent_containers/food/condiment.dm [hunk 2]
```diff
@@ -15,136 +15,138 @@
...
 		return
-	attack_self(mob/user as mob)
+
+/obj/item/reagent_containers/food/condiment/attack_self(mob/user as mob)
```
Comment (SplinterGP) [inline code/game/objects/items/reagent_containers/food/condiment.dm:21]:
```suggestion
/obj/item/reagent_containers/food/condiment/attack_self(mob/user)
```

File: code/game/objects/items/reagent_containers/food/condiment.dm [hunk 3]
```diff
@@ -15,136 +15,138 @@
...
 		return
-	attack(mob/M as mob, mob/user as mob, def_zone)
-		var/datum/reagents/R = src.reagents
 
-		if(!R || !R.total_volume)
-			to_chat(user, span_warning("The [src.name] is empty!"))
-			return 0
+/obj/item/reagent_containers/food/condiment/attack(mob/M as mob, mob/user as mob, def_zone)
```
Comment (SplinterGP) [inline code/game/objects/items/reagent_containers/food/condiment.dm:24]:
```suggestion
/obj/item/reagent_containers/food/condiment/attack(mob/M, mob/user, def_zone)
```

File: code/game/objects/items/reagent_containers/food/condiment.dm [hunk 4]
```diff
@@ -15,136 +15,138 @@
...
+	var/datum/reagents/R = src.reagents
```
Comment (SplinterGP) [inline code/game/objects/items/reagent_containers/food/condiment.dm:25]:
```suggestion
	var/datum/reagents/R = reagents
```

File: code/game/objects/items/reagent_containers/food/condiment.dm [hunk 5]
```diff
@@ -15,136 +15,138 @@
...
-		return 0
-
-	attackby(obj/item/I as obj, mob/user as mob)
-
-		return
+	return 0
 
-	afterattack(obj/target, mob/user , flag)
-		if(istype(target, /obj/structure/reagent_dispensers)) //A dispenser. Transfer FROM it TO us.
+/obj/item/reagent_containers/food/condiment/attackby(obj/item/I as obj, mob/user as mob)
```
Comment (SplinterGP) [inline code/game/objects/items/reagent_containers/food/condiment.dm:59]:
```suggestion
/obj/item/reagent_containers/food/condiment/attackby(obj/item/I, mob/user)
```

**Review by SplinterGP — CHANGES_REQUESTED** [review]

**Review by BraveMole — APPROVED** [review]

**Pull request merged**